src/cpp/core/system/Xdg.cpp

```diff
@@ -108,17 +109,53 @@ FilePath resolveXdgDir(const std::string& envVar,
 } // anonymous namespace
 
 FilePath userConfigDir(
-        const boost::optional<std::string>& user,
-        const boost::optional<FilePath>& homeDir)
+   const boost::optional<std::string>& user,
+   const boost::optional<FilePath>& homeDir)
 {
-   return resolveXdgDir("XDG_CONFIG_HOME", 
+   FilePath confDir = resolveXdgDir("XDG_CONFIG_HOME", 
 #ifdef _WIN32
          FOLDERID_RoamingAppData,
 #endif
          "~/.config",
          user,
          homeDir
    );
+
+   Error permError;
+   Error error = confDir.ensureDirectory();
+   if (!error)
+   {
+      // Test permissions here.
+      error = confDir.testWritePermissions();
```

Comment:
We call userConfigDir all the time and it's expected to be a fast, read-only operation -- probably not the right place to add a write test. This test should be done only once early at startup.

Reply:
Should we test this at session startup, server startup or both? We do read from here in the server/desktop code for crashpad settings, I think.